cmd/kubeadm/app/util/config/initconfiguration_test.go

```diff
@@ -78,6 +81,11 @@ func TestLoadInitConfigurationFromFile(t *testing.T) {
 					t.Error("Unexpected success")
 				}
 			} else {
+				// If the test run on the node with multiple cri endpoints configured and cri endpoint is not set in the InitConfiguration, there should be an error to detect the
```

Comment:
i think we should avoid having tests depend on host settings in all cases. this means ideally we should not test this scenario at all.

can we remove it and use the unknown socket?

Reply:
testcase of `v1beta3.partial2` wants to check that `ClusterConfiguration` can be loaded and finally converted to `InitConfiguration`, I didn't find a way to set the socket without the cfg of `InitConfiguration`. But without the set of cri socket, the testcase won't pass with multiple cri configured.

Reply:
why not add a stub InitConfiguration that has some user defined `criSocket: "foo"`

```
name:         "v1beta3.partial2",
			fileContents: bytes.Join([][]byte{
				cfgFiles["ClusterConfiguration_v1beta3"],
				cfgFiles["InitConfiguration_v1beta3"], // some stub with just socket?
			}, []byte(constants.YAMLDocumentSeparator)),
		},
```

it would still convert the two into a single init configuration.

Reply:
it seems InitConfiguration_v1beta3 just has the socket and that's what we need. we can then remove the ignoreErrStr field and check?

Reply:
> it would still convert the two into a single init configuration.

yes, that should work, but then it will not the scenario this testcase wants to cover? 

Originally, 

- partial1: only has the `InitConfiguration`
- partial2: only has the `ClusterConfiguration`
- full: has `InitConfiguration`, `ClusterConfiguration` and addons.

If we do that like that, it would be,
- partial1: only has the `InitConfiguration`
- **partial2: both the `ClusterConfiguration` and `InitConfiguration`**
- full: has `InitConfiguration`, `ClusterConfiguration` and addons.

I think this will change the intention of the testcases, I am not sure about that.

@pacoxu @SataQiu  wdyt?

Reply:
A tricky fix is like below:

1. change defaultKnownCRISockets to DefaultKnownCRISockets so that it can be changed in this UT.
2. add `runtime.DefaultKnownCRISockets = []string{constants.UnknownCRISocket}` before t.Run in this test func.

Reply:
that is indeed tricky, then we need to set it back at the end of test. I am not a fan of that. :)

Reply:
> If we do that like that, it would be

how about - we do we it like i proposed snd then we can inspect obj to ensure that it contains what we want.
we can pass a function "validate" per test case which if not nil can be used to check if field "foo" exists?
if "foo" exists it came from a ClusterConfiguration.